src/driver/linux_onload/ossock_calls.c

```diff
@@ -112,12 +114,18 @@ oo_fd_replace_file(struct file* old_filp, struct file* new_filp,
   }
 
   rcu_read_lock();
-  if( lookup_fdget_rcu(old_fd) != old_filp ) {
+  tmp_filp = lookup_fdget_rcu(old_fd);
+  if( tmp_filp != old_filp ) {
+    if( tmp_filp )
+      fput(tmp_filp);
     rcu_read_unlock();
     spin_unlock(&current->files->file_lock);
```

Comment:
Since we're here...

Do you know where we call the corresponding `spin_lock(&current->files->file_lock)`? Maybe it should call `task_unlock(current);` instead?

Reply:
`task_unlock` is `spin_unlock(&p->alloc_lock)`, which differ from `spin_unlock(&p->files->file_lock)`.

I guess you've found a real bug, because this error path was never tested.

Reply:
See #206